pkg/gui/presentation/branches.go

```diff
@@ -143,6 +149,24 @@ func GetBranchTextStyle(name string) style.TextStyle {
 	}
 }
 
+func (m *colorMatcher) match(name string) (*style.TextStyle, bool) {
+	if m.isRegex {
+		for pattern, style := range m.patterns {
+			if matched, _ := regexp.MatchString("^"+pattern+"$", name); matched {
```

Comment:
I find the implicit anchoring surprising, and would remove it. If users want it, they can easily add it themselves. For the record, we don't use it for [commit prefix patterns](https://github.com/jesseduffield/lazygit/blob/5e9ba82d7bcdfdc873015f24a0817378e2fa4788/pkg/gui/controllers/helpers/working_tree_helper.go#L160) either.

Reply:
The reason I did this was because the config value `'IDEA-\d': 'blue'` had this behavior:
<img width="400" alt="Screenshot 2025-01-04 at 3 35 48 PM" src="https://github.com/user-attachments/assets/f2bcdd98-d795-465b-9409-fb1baf46e3aa" />

Which seemed like the incorrect behavior to me, my expectation was that it should only highlight the one with the single digit in it. But I do agree that it's better to be explicit and stick with similar behavior on other parts of the app. I'll change this